lib/manager/cargo/update.js

```diff
@@ -1,39 +1,123 @@
-const { getDep, getInlineTableDep, getTableDep } = require('./extract');
+const _ = require('lodash');
+const toml = require('toml');
 
 module.exports = {
   updateDependency,
 };
 
-function updateDependency(currentFileContent, upgrade) {
-  logger.trace({ config: upgrade }, 'cargo.updateDependency()');
-  const { depName, depType, lineNumber, newValue } = upgrade;
-  const lines = currentFileContent.split('\n');
-  let dep;
-  let currentValue;
-  let versionLineNumber = lineNumber;
-  const line = lines[lineNumber];
-  if (depType === 'normal') {
-    dep = getDep(line);
-  } else if (depType === 'inlineTable') {
-    dep = getInlineTableDep(line);
-  } else if (depType === 'standardTable' && line) {
-    const match = line.match('dependencies[.]');
-    // Might need handling comments in getTableDep
-    dep = getTableDep(match, lines, lineNumber);
-    if (dep) {
-      versionLineNumber = dep.versionLineNumber;
-    }
+// TODO: Maybe factor out common code from pipenv.updateDependency and poetry.updateDependency
```

Comment:
See https://github.com/renovatebot/renovate/issues/3508

It does not need to be done now though